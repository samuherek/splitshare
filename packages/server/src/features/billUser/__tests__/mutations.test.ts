import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';

describe('bill invite mutations', () => {
  test('createInvite: mutation should create and return a bill invite', async () => {
    const CREATE_INVITE = gql`
      mutation createInvite($input: CreateInviteInput!) {
        createInvite(input: $input) {
          id
          email
          pending
          createdAt
          updatedAt
          deletedAt
        }
      }
    `;

    const { mutate } = await createTestServer({
      user: { id: 1 },
      models: {
        BillInvite: {
          createOne: jest.fn(() => ({
            id: 1,
            email: 'email@email.com',
            pending: true,
            createdAt: new Date('2000-01-01'),
            updatedAt: new Date('2000-01-01'),
            deletedAt: null,
          })),
        },
      },
    });

    const res = await mutate({
      mutation: CREATE_INVITE,
      variables: { input: { email: 'email@email.com', billId: 1 } },
    });

    expect(res).toMatchSnapshot();
  });
});
