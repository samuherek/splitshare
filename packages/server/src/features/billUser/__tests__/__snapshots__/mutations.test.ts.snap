// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bill invite mutations createInvite: mutation should create and return a bill invite 1`] = `
Object {
  "data": Object {
    "createInvite": Object {
      "createdAt": "2000-01-01",
      "deletedAt": null,
      "email": "email@email.com",
      "id": "1",
      "pending": true,
      "updatedAt": "2000-01-01T00:00:00.000Z",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
